appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -1049,9 +1083,10 @@ Blockly.LexicalVariable.getEventParam = function (block) {
     var name = prefixPair[1];
     var child = block;
     var parent = block.getParent();
+    var type = parent.type;
```

Comment:
This code is walking up the block tree trying to find the appropriate lexical scope. This evaluation of `parent.type` needs to happen within the while loop otherwise `type` is always the immediate parent's type rather than the parent of the currently evaluated block within the loop.